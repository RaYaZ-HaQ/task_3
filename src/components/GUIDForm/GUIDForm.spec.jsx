import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Form from ".";

describe("Form tests", () => {
  it("should match snapshot", () => {
    const { container } = render(<Form />);

    expect(container).toMatchSnapshot();
  });

  it("should disable input and button while loading", async () => {
    render(<Form />);
    const user = userEvent.setup();

    let input = screen.getByRole("textbox");
    await user.type(input, "sample text");
    let button = screen.getByRole("button");
    await user.click(button);

    const loadingIcon = await screen.findByTestId("loading-icon");
    expect(loadingIcon).toBeTruthy();
    expect(screen.getByRole("textbox").getAttributeNames()).toContain(
      "disabled"
    );
    expect(screen.getByRole("button").getAttributeNames()).toContain(
      "disabled"
    );
  });

  it("should validate a correct guid", async () => {
    render(<Form />);
    const user = userEvent.setup();

    const input = screen.getByRole("textbox");
    await user.type(input, "c56a4180-65aa-42ec-a945-5fd21dec0538");
    const button = screen.getByRole("button");
    await user.click(button);

    const successIcon = await screen.findByTestId(
      "check-icon",
      {},
      { timeout: 3000 }
    );
    expect(successIcon).toBeTruthy();
  });

  it("should show an error for a invalid guid", async () => {
    render(<Form />);
    const user = userEvent.setup();

    const input = screen.getByRole("textbox");
    await user.type(input, "invalid-guid");
    const button = screen.getByRole("button");
    await user.click(button);

    const successIcon = await screen.queryByTestId("check-icon");
    expect(successIcon).toBeFalsy();
    const errorPill = await screen.findByText(
      "GUID Invalid",
      {},
      { timeout: 3000 }
    );
    expect(errorPill).toBeTruthy();
  });
});
