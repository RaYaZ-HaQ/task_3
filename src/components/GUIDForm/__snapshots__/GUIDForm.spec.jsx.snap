// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form tests should match snapshot 1`] = `
<div>
  <form
    class="flex-container-column spacing-20"
  >
    <div
      class="flex-container items-center"
    >
      <label>
        Enter a valid GUID here.
      </label>
    </div>
    <input
      class="text-field"
      id="guid"
      name="guid"
      type="text"
    />
    <button
      class="button flex-container items-center"
    >
      <span>
        Submit
      </span>
    </button>
  </form>
</div>
`;
